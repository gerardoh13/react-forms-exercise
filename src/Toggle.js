import React from "react";

const Toggle = ({ toggleApp }) => {
  const handleChange = (e) => {
    toggleApp(e.target.value);
  };
  return (
    <div className="row mb-3 mt-5">
      <div className="col">
        <input
          type="radio"
          className="btn-check"
          name="display"
          id="boxes"
          value="boxes"
          defaultChecked
          onChange={handleChange}
        />
        <label className="btn btn-outline-secondary" htmlFor="boxes">
          Boxes
        </label>
      </div>
      <div className="col">
        <input
          type="radio"
          className="btn-check"
          name="display"
          id="todo"
          value="todo"
          onChange={handleChange}
        />
        <label className="btn btn-outline-secondary" htmlFor="todo">
          Todos
        </label>
      </div>
    </div>
  );
};

export default Toggle;
