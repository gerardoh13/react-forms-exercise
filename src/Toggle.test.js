
import React from "react";
import { render } from "@testing-library/react";
import Toggle from "./Toggle";

it("renders without crashing", function() {
  render(<Toggle />);
});

it("matches snapshot", function() {
  const { asFragment } = render(<Toggle />);
  expect(asFragment()).toMatchSnapshot();
});