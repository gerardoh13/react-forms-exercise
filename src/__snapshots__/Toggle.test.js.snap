// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="row mb-3 mt-5"
  >
    <div
      class="col"
    >
      <input
        checked=""
        class="btn-check"
        id="boxes"
        name="display"
        type="radio"
        value="boxes"
      />
      <label
        class="btn btn-outline-secondary"
        for="boxes"
      >
        Boxes
      </label>
    </div>
    <div
      class="col"
    >
      <input
        class="btn-check"
        id="todo"
        name="display"
        type="radio"
        value="todo"
      />
      <label
        class="btn btn-outline-secondary"
        for="todo"
      >
        Todos
      </label>
    </div>
  </div>
</DocumentFragment>
`;
